src/lib.rs

```diff
@@ -137,3 +138,80 @@ pub fn extract_release_archive(
 
     Ok(())
 }
+
+async fn fetch_program(
+    name: &str,
+    version: &str,
+    solana_root_path: &Path,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let so_filename = format!("spl_{}-{}.so", name.replace('-', "_"), version);
+    let so_path = solana_root_path.join(&so_filename);
+
+    if !so_path.exists() {
+        info!("Downloading {} {}", name, version);
+        let url = format!(
+            "https://github.com/solana-labs/solana-program-library/releases/download/{}-v{}/{}",
+            name, version, so_filename
+        );
+
+        download_to_temp(&url, &so_path)
+            .await
+            .map_err(|err| format!("Unable to download {url}. Error: {err}"))?;
+    }
+
+    Ok(())
+}
+
+pub async fn fetch_spl(solana_root_path: &Path) -> Result<(), Box<dyn std::error::Error>> {
+    let mut genesis_args = vec![];
+
+    let programs = vec![
+        (
+            "token",
+            "3.5.0",
+            "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
+            "BPFLoader2111111111111111111111111111111111",
+        ),
+        (
+            "token-2022",
+            "0.9.0",
+            "TokenzQdBNbLqP5VEhdkAS6EPFLC1PHnBqCXEpPxuEb",
+            UPGRADEABLE_LOADER,
+        ),
+        (
+            "associated-token-account",
+            "1.1.2",
+            "ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL",
+            "BPFLoader2111111111111111111111111111111111",
+        ),
+    ];
```

Comment:
These seem to be missing memo v1, memo v3, and feature-proposal: https://github.com/anza-xyz/agave/blob/deec1f450d1f2eae40158f481a37f1e8be79ca03/fetch-spl.sh#L49

Was that intentional?

Reply:
ahhhh shoot. i had some url issues with these. and took them out but forgot to put them back in once i figured out the issue. smh. will fix